src/nsrt_learning/strips_learning/gen_to_spec_learner.py

```diff
@@ -62,14 +63,14 @@ def _learn(self) -> List[PartialNSRTAndDatastore]:
             for segment in seg_traj:
                 parameterized_options.add(segment.get_option().parent)
 
-        # Set up the param_opt_to_general_pnad and param_opt_to_nec_pnads
-        # dictionaries.
+        # Set up the and param_opt_to_nec_pnads dictionary.
         for param_opt in parameterized_options:
-            pnad = self._initialize_general_pnad_for_option(param_opt)
-            param_opt_to_general_pnad[param_opt] = pnad
+            pnad = self._create_general_pnad_for_option(param_opt)
```

Comment:
why are you creating this `pnad` variable but then never using it? why do we need to call this function here at all? can we just remove all this initializing?